tests/integration/pilot/common/routing.go

```diff
@@ -618,46 +626,43 @@ func trafficLoopCases(apps *EchoDeployments) []TrafficTestCase {
 	return cases
 }
 
-func gatewayCases(apps *EchoDeployments) []TrafficTestCase {
-	cases := []TrafficTestCase{}
-	podAFQDN := apps.PodA[0].Config().FQDN()
-
-	destinationSets := []echo.Instances{
-		apps.PodA,
-		apps.VM,
-		apps.Naked,
-		apps.Headless,
-		apps.External,
+func gatewayCases() []TrafficTestCase {
+	templateParams := func(protocol protocol.Instance, dests echo.Services) map[string]interface{} {
+		dest, portN, cred := dests[0][0], 80, ""
```

Comment:
why do we use `dests[0][0]`? Shouldn't we allow any dest, not just the first?

Reply:
is it because the host doesn't matter? if so it may make sense to use a dummy value instead of picking an arbitrary one, as its a bit confusing

Reply:
We use `[0][0]` because we're not using `RunForN`, so we only care about the first service. Then for the VS host, the `Instances` share an FQDN. 

I want to clean up these interfaces a bit to hide all the `[0]`s, but just getting the tests to pass for now.